client/src/templates/Challenges/projects/SolutionForm.tsx

```diff
@@ -10,25 +9,37 @@ import {
   pythonProject
 } from '../../../../utils/challengeTypes';
 
-const propTypes = {
-  challengeType: PropTypes.number,
-  description: PropTypes.string,
-  isSubmitting: PropTypes.bool,
-  onSubmit: PropTypes.func.isRequired,
-  t: PropTypes.func.isRequired,
-  updateSolutionForm: PropTypes.func.isRequired
-};
-
-export class SolutionForm extends Component {
-  constructor(props) {
+interface SubmitProps {
+  isShouldCompletionModalOpen: boolean;
+}
+
+interface FormProps {
+  challengeType: number;
+  description: string;
+  isSubmitting: boolean;
+  onSubmit: (arg0: SubmitProps) => void;
+  t: (arg0: string) => string;
+  updateSolutionForm: (arg0: Record<string, unknown>) => void;
+  placeholders: (arg0: Record<string, unknown>) => void;
+}
```

Comment:
```suggestion
interface FormProps extends WithTranslation {
  challengeType: number;
  description: string;
  isSubmitting: boolean;
  onSubmit: (arg0: SubmitProps) => void;
  updateSolutionForm: (arg0: Record<string, unknown>) => void;
}
```
A few things here: `WithTranslation` gives us a `t` prop.  Also, `isSubmitting` hasn't been passed for years.  I'm not 100% sure what it did, but it's not doing it.  Finally, `placeholders` isn't a prop.